src/sqlfluff/rules/L033.py

```diff
@@ -26,15 +26,38 @@ class Rule_L033(BaseRule):
 
     """
 
-    def _eval(self, segment, raw_stack, **kwargs):
+    def _eval(self, segment, raw_stack, dialect, **kwargs):
         """Look for UNION keyword not immediately followed by DISTINCT or ALL.
 
         Note that UNION DISTINCT is valid, rule only applies to bare UNION.
         The function does this by looking for a segment of type set_operator
         which has a UNION but no DISTINCT or ALL.
+
+        Note only some dialects have concept of UNION DISTINCT, so rule is only
+        applied to dialects that are known to support this syntax.
         """
+        if dialect.name not in ["ansi", "bigquery", "hive", "mysql"]:
+            return LintResult()
+
         if segment.is_type("set_operator"):
-            if "UNION" in segment.raw.upper() and not (
+            if "union" in segment.raw and not (
```

Comment:
Is the `.raw` property guaranteed to be lower case? If you're removing `.upper()` here, could it also be removed from line 44 below?

Reply:
It's the raw casing - so whatever is in the source.

So I'm using this to say "it's it's lowercase in source, then enter the `distinct` as lowercase." For Line 44 I don't really care about the case, so keep it as case-less matching.

This will match in the unlikely event of:

```sql
SELECT 1
union ALL
SELECT 2
```